artifactory/services/groups.go

```diff
@@ -0,0 +1,89 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	rthttpclient "github.com/jfrog/jfrog-client-go/artifactory/httpclient"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+type Group struct {
+	Name            string `json:"name,omitempty"`
+	Description     string `json:"description,omitempty"`
+	AutoJoin        bool   `json:"autoJoin,omitempty"`
+	AdminPrivileges bool   `json:"adminPrivileges,omitempty"`
+	Realm           string `json:"realm,omitempty"`
+	RealmAttributes string `json:"realmAttributes,omitempty"`
+}
+type GroupService struct {
+	client     *rthttpclient.ArtifactoryHttpClient
+	ArtDetails auth.ServiceDetails
+}
+
+func NewGroupService(client *rthttpclient.ArtifactoryHttpClient) *GroupService {
+	return &GroupService{client: client}
+}
+func (gs *GroupService) SetArtifactoryDetails(rt auth.ServiceDetails) {
+	gs.ArtDetails = rt
+}
+
+func (gs *GroupService) GetGroup(name string) (*Group, error) {
+	httpDetails := gs.ArtDetails.CreateHttpClientDetails()
+	url := fmt.Sprintf("%sapi/security/groups/%s", gs.ArtDetails.GetUrl(), name)
+	res, body, _, err := gs.client.SendGet(url, true, &httpDetails)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode > 204 {
+		return nil, fmt.Errorf("%d %s: %s", res.StatusCode, res.Status, string(body))
+	}
+	var group Group
+	if err := json.Unmarshal(body, &group); err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	return &group, nil
+}
+
+func (gs *GroupService) CreateOrUpdateGroup(group Group) error {
+	httpDetails := gs.ArtDetails.CreateHttpClientDetails()
+	content, err := json.Marshal(group)
+	if err != nil {
+		return err
+	}
+
+	httpDetails.Headers = map[string]string{
+		"Content-Type": "application/json",
+		"Accept":       "application/json",
+	}
+	//
+	url := fmt.Sprintf("%sapi/security/groups/%s", gs.ArtDetails.GetUrl(), group.Name)
+	resp, body, err := gs.client.SendPut(url, content, &httpDetails)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode > 204 {
+		return fmt.Errorf("%d %s: %s", resp.StatusCode, resp.Status, string(body))
```

Comment:
See my comment in the closed PR about this - let's stuck to the existing convention.